lib/matplotlib/backends/qt_compat.py

```diff
@@ -109,9 +109,18 @@
         except:
             res = 'QVariant API v2 specification failed. Defaulting to v1.'
             verbose.report(cond + res, 'helpful')
+    if QT_API == QT_API_PYQT5:
+        try:
+            from PyQt5 import QtCore, QtGui, QtWidgets
+            _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
+        except ImportError:
+            # fell through, tried PyQt5, failed fall back to PyQt4
+            QT_API = rcParams['backend.qt4']
```

Comment:
Ah, the logic has gotten crossed here. You can fall through this code path as the 'right' thing to do if we had to guess about qt4vqt5 above